core/src/main/java/org/mskcc/cbio/portal/dao/DaoReferenceGenome.java

```diff
@@ -0,0 +1,299 @@
+                + "`build_name`=?, `genome_size`=?, `URL`=?, "
+                + " `release_date`=? WHERE `reference_genome_id`=?");
+            pstmt.setString(1, referenceGenome.getSpecies());
+            pstmt.setString(2, referenceGenome.getGenomeName());
+            pstmt.setString(3, referenceGenome.getBuildName());
+            pstmt.setLong(4, referenceGenome.getGenomeSize());
+            pstmt.setString(5, referenceGenome.getUrl());
+            pstmt.setDate(6, new java.sql.Date(referenceGenome.getReleaseDate().getTime()));
+            pstmt.setInt(7, referenceGenome.getReferenceGenomeId());
+            rows += pstmt.executeUpdate();
+            // update reference cache
+            reCache();
+            return rows;
+        } catch (SQLException e) {
+            throw new DaoException(e);
+        } finally {
+            JdbcUtil.closeAll(DaoReferenceGenome.class, con, pstmt, rs);
+        }
+    }
+
+    /**
+     * Retrieve reference genome by internal DB ID
+     * @param internalId   Reference Genome internal DB ID
+     * @throws DaoException Database Error.
+     */
+
+    public static ReferenceGenome getReferenceGenomeByInternalId(int internalId) throws DaoException {
+        ReferenceGenome genome = byGenomeInternalId.get(internalId);
+        return genome;
+    }
+
+    /**
+     * Retrieve reference genome by genome build name
+     * @param buildName   Reference Genome build name
+     * @throws DaoException Database Error.
+     */
+
+    public static ReferenceGenome getReferenceGenomeByBuildName(String buildName) throws DaoException {
+        return byGenomeBuild.get(buildName);
+    }
+
+
+    /**
+     * Retrieve reference genome by genome build name
+     * @param genomeName   Reference Genome build name
+     * @throws DaoException Database Error.
+     */
+
+    public static ReferenceGenome getReferenceGenomeByGenomeName(String genomeName) throws DaoException {
+        //return byGenomeName.get(genomeName);
+        Connection con = null;
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;
+        ReferenceGenome referenceGenome = null;
+        try {
+            con = JdbcUtil.getDbConnection(DaoReferenceGenome.class);
+            pstmt = con.prepareStatement("SELECT * FROM reference_genome WHERE `name` = ?");
+            pstmt.setString(1,genomeName);
+            rs = pstmt.executeQuery();
+            if (rs.next()) {
+                referenceGenome = extractReferenceGenome(rs);
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            JdbcUtil.closeAll(DaoReferenceGenome.class, con, pstmt, rs);
+            return referenceGenome;
+        }
+    }
+
+    /**
+     * Retrieve reference genome of interest by genome name or genome assembly name
+     * @param name   Name of Reference Genome or Genome Assembly
+     * @throws DaoException Database Error.
+     */
+    @Deprecated
```

Comment:
all right, will remove it.

Reply:
gave it a second thought. We might want to remove deprecated tag, since we already decided we are not going to support multiple species per portal installation, and most portal will be for human. Let's keep it, I would say.